### PR TITLE
Trace analytics ui bug fixes in custom sources

### PR DESCRIPTION
**Pull request by ps48**
### Description
Trace analytics UI bug fixes in custom sources

### Issues Resolved
* Fix window scroll on top when selecting a filter from services list
* Fix refresh in traces page when search is made using "Shift+Enter"
* Updated service map empty state behaviour - on click load latency metrics

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by paulstn — APPROVED** [review]

File: public/components/trace_analytics/requests/queries/traces_queries.ts [hunk 1]
```diff
@@ -568,7 +568,7 @@ export const getCustomIndicesTracesQuery = (
         must_not: [],
       },
     },
-    ...(sort && { sort: [{ [sort.field]: { order: sort.direction } }] }),
+    ...(sort ? { sort: [{ [sort.field]: { order: sort.direction } }] } : {}),
```
Comment (joshuali925) [inline public/components/trace_analytics/requests/queries/traces_queries.ts:571]: is this change needed?

**Review by TackAdam — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

File: public/components/trace_analytics/requests/queries/traces_queries.ts (hunk 1 above)
Comment (ps48) [reply public/components/trace_analytics/requests/queries/traces_queries.ts:571]: TBH not needed, I was seeing some issues with search made with keyboard hit `shift+enter`. The root cause was this: `onSearch={(searchQuery) => props.refresh(undefined, searchQuery)}`. But this is part of experimental feature and will be changed when it becomes GA.

Comment (ps48) [reply public/components/trace_analytics/requests/queries/traces_queries.ts]: reverted 😄
